pkg/component/rsyslogrelpconfigcleaner/rsyslog_relp_config_cleaner.go

```diff
@@ -0,0 +1,298 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package rsyslogrelpconfigcleaner
+
+import (
+	"context"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extension-shoot-rsyslog-relp/pkg/constants"
+)
+
+const managedResourceName = "extension-" + constants.ServiceName + "-configuration-cleaner-shoot"
```

Comment:
In the registry-cache extension we don't suffix the MR with `-shoot`. The MR already has spec.class=nil which should mean .spec.class=nil.

```suggestion
const managedResourceName = "extension-" + constants.ServiceName + "-configuration-cleaner"
```

But no strong opinion. If you want to keep it like this, let's don't change it.